### PR TITLE
allow to pass metadata from transport layer with the stanza erlang msg

### PR DESCRIPTION
**Pull request by michalwski**
Currently there is only `recv_timestamp` in the metadata sent from transport process to the client using escalus. The timestamp is taken short after the transport receives the packet.

### CODE REVIEW

**Comment by michalwski** [comment]
This is still in progress. Some how 2 test cases, one in global_dist and one in mam are always failing with these changes... More details can be found in michalwski/MongooseIM#110

File: src/escalus_connection.erl [hunk 1]
```diff
@@ -177,25 +183,44 @@ get_stanza(Conn, Name) ->
...
-get_stanza(#client{rcv_pid = Pid, jid = Jid}, Name, Timeout) ->
+get_stanza(Client, Name, Timeout) ->
+    case get_stanza_safe(Client, Timeout) of
+        {error, timeout} ->
+            throw({timeout, Name});
+        {Stanza, _} ->
+            Stanza
+    end.
+
+-spec get_stanza_with_metadata(client(), any(), timeout()) -> {exml_stream:element(), map()}.
```
Comment (arkgil) [inline src/escalus_connection.erl]: Would it be possible to spec metadata map more precisely? I see only `recv_timeout` key, so I guess we could add a spec for that. Is it possible to somehow add own metadata keys?

Comment (michalwski) [reply src/escalus_connection.erl]: The problem with map spec is that it's supported from Erlang/OTP 19 while escalus is officially tested with Erlang/OTP 18 to 20. I think more precise type could be added when we ditch support for 18.

Comment (michalwski) [reply src/escalus_connection.erl]: Currently it's not possible to add your own metadata. How would you see it?

Comment (fenek) [reply src/escalus_connection.erl]: @michalwski A spec could be commented out for now but still usable as a reference for a developer. :)

**Comment by michalwski** [comment]
I consider it done and working. Test run with MongooseIM's tests was done on my fork michalwski/MongooseIM#110

**Pull request merged**